utils/text_formatting.test.jsx

```diff
@@ -213,6 +213,21 @@ describe('parseSearchTerms', () => {
             input: 'from:someone someword "some phrase" in:somechannel',
             expected: ['someword', 'some phrase'],
         },
+        {
+            description: 'with negative search flags after',
+            input: 'someword "some phrase" -from:someone -in:somechannel',
+            expected: ['someword', 'some phrase'],
+        },
+        {
+            description: 'with negative search flags before',
+            input: '-from:someone -in: channel someword "some phrase"',
+            expected: ['someword', 'some phrase'],
+        },
+        {
+            description: 'with negative search flags before and after',
+            input: '-from:someone someword "some phrase" -in:somechannel',
+            expected: ['someword', 'some phrase'],
+        },
```

Comment:
Should these tests include a case with `-on` like in the screenshot?

Reply:
Actually, the date related flags (`on`, `before` and `after`) were not being excluded from the highlighted terms, and it would have been possible for someone to search with a date filter and get a result with that exact same date filter as a highlighted message.

I've modified the regexes to exclude those search flags and added the related tests. Thanks for catching this one!